src/nvim/os/fs.c

```diff
@@ -164,21 +164,25 @@ int os_nodetype(const char *name)
 ///
 /// @param[in]  name     Name of the executable.
 /// @param[out] abspath  Path of the executable, if found and not `NULL`.
+/// @param[in] use_path  If 'false', only check if "name" is executable
 ///
 /// @return `true` if `name` is executable and
 ///   - can be found in $PATH,
 ///   - is relative to current dir or
 ///   - is absolute.
 ///
 /// @return `false` otherwise.
-bool os_can_exe(const char_u *name, char_u **abspath)
+bool os_can_exe(const char_u *name, char_u **abspath, bool use_path)
   FUNC_ATTR_NONNULL_ARG(1)
 {
-  // If it's an absolute or relative path don't need to use $PATH.
-  if (path_is_absolute_path(name) ||
+  // when use_path is false or if it's an absolute or relative path don't
+  // need to use $PATH.
+  if (!use_path || path_is_absolute_path(name) ||
      (name[0] == '.' && (name[1] == '/' ||
                         (name[1] == '.' && name[2] == '/')))) {
-    if (is_executable(name)) {
+    // There must be a path separator, files in the current directory
+    // can't be executed
+    if (gettail_dir(name) != name && is_executable(name)) {
```

Comment:
Turns out this hasn't been a problem at all :) I just needed to adjust the tests for the third argument of `os_can_exe`. The main problem was using `false` as a third argument [here](https://github.com/neovim/neovim/pull/4678/files#diff-deb695d507653839019814f54242d47bR11612), where it really needs to be `true`. That _really_ broke some stuff.